packages/types/src/node.ts

```diff
@@ -64,6 +64,16 @@ export namespace Node {
     reset?(): Promise<void>;
   }
 
+  export interface ILockService {
+    get(path: string): Promise<Lock>;
```

Comment:
`checkIfLockExists`?

Reply:
See above comment; we do not need a method to check. We should just attempt to acquire a lock and execute code if successful. Implementations can decide to attempt to acquire the lock multiple times as business logic necessitates...

Reply:
...meaning, if they need retry logic.